lib/auth/auth.go

```diff
@@ -1839,7 +1854,11 @@ func (a *Server) ExtendWebSession(ctx context.Context, req WebSessionReq, identi
 		roles = append(roles, accessRequest.GetRoles()...)
 		roles = apiutils.Deduplicate(roles)
 		accessRequests = apiutils.Deduplicate(append(accessRequests, req.AccessRequestID))
-		requestedResourceIDs = accessRequest.GetRequestedResourceIDs()
+
+		if len(allowedResourceIDs) > 0 && len(accessRequest.GetRequestedResourceIDs()) > 0 {
```

Comment:
I don't quite understand what this check is enforcing. Can you add a comment to clarify?